storage/redis_cluster.go

```diff
@@ -28,6 +33,55 @@ type RedisCluster struct {
 	IsCache   bool
 }
 
+func clusterConnectionIsOpen(cluster *rediscluster.RedisCluster) bool {
+	// check if at least one of cluster handles has an alive connection
+	for item := range cluster.Handles.Iter() {
+		conn := item.Val.(*rediscluster.RedisHandle).GetRedisConn()
+		isOpen := conn.Err() == nil
+		conn.Close()
```

Comment:
not really sure how this works... but if `conn.Err != nil`, can you safely call `conn.Close`?

Reply:
@asoorm it comes to https://github.com/TykTechnologies/tyk/blob/master/vendor/github.com/lonelycode/redigocluster/rediscluster/redishandle.go#L58
which calls https://github.com/TykTechnologies/tyk/blob/master/vendor/github.com/garyburd/redigo/redis/pool.go#L157 
as they say:
```
// Get gets a connection. The application must close the returned connection.
// This method always returns a valid connection so that applications can defer
// error handling to the first use of the connection. If there is an error
// getting an underlying connection, then the connection Err, Do, Send, Flush
// and Receive methods return that error.
```
so we close it